File: paper/src/main/java/com/al3x/housing2/Data/LayoutData.java
package com.al3x.housing2.Data;

import com.al3x.housing2.Instances.Layout;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
public class LayoutData {
    private String name;
    private String icon;
    private String description;
    private List<String> inventory;
    private List<String> hotbar;
    private List<String> armor;
    private String offhand;

    public LayoutData() {

    }

    public LayoutData(String name, String icon, String description, List<String> inventory, List<String> hotbar, List<String> armor, String offhand) {
        this.name = name;
        this.icon = icon;
        this.description = description;
        this.inventory = inventory;
        this.hotbar = hotbar;
        this.armor = armor;
        this.offhand = offhand;
    }

    public static List<LayoutData> fromList(List<Layout> layouts) {
        List<LayoutData> layoutDatas = new ArrayList<>();
        for (Layout layout : layouts) {
            layoutDatas.add(layout.toData());
        }
        return layoutDatas;
    }

    public static List<Layout> toList(List<LayoutData> layoutDatas) {
        List<Layout> layouts = new ArrayList<>();
        for (LayoutData layoutData : layoutDatas) {
            layouts.add(new Layout(
                layoutData.getName(),
                layoutData.getIcon(),
                layoutData.getDescription(),
                layoutData.getInventory(),
                layoutData.getHotbar(),
                layoutData.getArmor(),
                layoutData.getOffhand()
            ));
        }
        return layouts;
    }
}

File: src/main/java/com/al3x/housing2/Instances/HousingData/LayoutData.kt
package com.al3x.housing2.Instances.HousingData

import com.al3x.housing2.Instances.Layout

data class LayoutData (
    var name: String,
    var icon: String,
    var description: String,
    var inventory: List<String>,
    var hotbar: List<String>,
    var armor: List<String>,
    var offhand: String?
) {
    companion object {
        fun fromList(layouts: List<Layout>): List<LayoutData> {
            val layoutDatas = mutableListOf<LayoutData>()
            for (layout in layouts) {
                layoutDatas.add(layout.toData())
            }
            return layoutDatas
        }

        fun toList(layoutDatas: List<LayoutData>): List<Layout> {
            val layouts = mutableListOf<Layout>()
            for (layoutData in layoutDatas) {
                layouts.add(Layout(layoutData.name, layoutData.icon, layoutData.description, layoutData.inventory, layoutData.hotbar, layoutData.armor, layoutData.offhand))
            }
            return layouts
        }
    }
}